src/Data/Algorand/Transaction.hs

```diff
@@ -118,29 +113,60 @@ data TransactionType
   deriving (Eq, Generic, Show)
 
 -- | Constants for @OnComplete@.
---
-type OnComplete = Text
-
-onCompleteNoOp :: OnComplete
-onCompleteNoOp = "call"
-
-onCompleteCreate :: OnComplete
-onCompleteCreate = "create"
-
-onCompleteOptIn :: OnComplete
-onCompleteOptIn = "optin"
-
-onCompleteCloseOut :: OnComplete
-onCompleteCloseOut = "closeout"
-
-onCompleteClearState :: OnComplete
-onCompleteClearState = "clear"
-
-onCompleteUpdateApplication :: OnComplete
-onCompleteUpdateApplication = "update"
-
-onCompleteDeleteApplication :: OnComplete
-onCompleteDeleteApplication = "delete"
+data OnComplete
+  = OnCompleteNoOp
+  | OnCompleteOptIn
+  | OnCompleteCloseOut
+  | OnCompleteClearState
+  | OnCompleteUpdateApplication
+  | OnCompleteDeleteApplication
+  deriving (Eq, Show)
```

Comment:
Maybe it's better to define `Show` manually?
```haskell
instance Show OnComplete where
  show OnCompleteNoOp = "noop"
  ...
```

Reply:
Added show instance manually now.